jupyterhub/handlers/base.py

```diff
@@ -1448,24 +1449,18 @@ def non_get(self, user_name, user_path):
     delete = non_get
 
     @web.authenticated
+    @needs_scope("access:servers")
     async def get(self, user_name, user_path):
         if not user_path:
             user_path = '/'
         current_user = self.current_user
-
-        if (
-            current_user
-            and current_user.name != user_name
-            and current_user.admin
-            and self.settings.get('admin_access', False)
-        ):
-            # allow admins to spawn on behalf of users
+        if user_name != current_user.name:
```

Comment:
A check if current user is truthy is removed, is it safe?

Reply:
Yes, the `@web.authenticated` decorator ensures current_user is truthy, or the method won't be called.